test/e2e/allowed_source_ranges_test.go

```diff
@@ -0,0 +1,221 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// TestAllowedSourceRanges creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type and verifies that the
+// operator correctly configures the associated load balancer and updates its
+// configuration when the ingresscontroller's allowed source ranges are changed.
+func TestAllowedSourceRanges(t *testing.T) {
+	t.Parallel()
+
+	if infraConfig.Status.PlatformStatus == nil {
+		t.Skip("test skipped on nil platform")
+	}
+	supportedPlatforms := map[configv1.PlatformType]struct{}{
+		configv1.AWSPlatformType:   {},
+		configv1.AzurePlatformType: {},
+		configv1.GCPPlatformType:   {},
+	}
+	if _, supported := supportedPlatforms[infraConfig.Status.PlatformStatus.Type]; !supported {
+		t.Skipf("test skipped on platform %q", infraConfig.Status.PlatformStatus.Type)
+	}
+
+	validCIDR := "127.0.0.0/8"
+	invalidCIDR := "127.0.0.1"
+	// Try creating an ingresscontroller with an invalid CIDR
+	t.Logf("Trying to create ingresscontroller with spec.endpointPublishingStrategy.loadBalancer.allowedSourceRanges set to an invalid CIDR range: %s", invalidCIDR)
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "sourcerange"}
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newLoadBalancerController(name, domain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		AllowedSourceRanges: []operatorv1.CIDR{operatorv1.CIDR(invalidCIDR)},
+		DNSManagementPolicy: operatorv1.ManagedLoadBalancerDNS,
+	}
+	if err := kclient.Create(context.TODO(), ic); err == nil {
+		assertIngressControllerDeleted(t, kclient, ic)
+		t.Fatalf("expected ingresscontroller creation to be failed with invalid CIDR: %s", invalidCIDR)
+	}
+
+	// Create an ingresscontroller with a valid CIDR
+	t.Logf("Creating ingresscontroller with spec.endpointPublishingStrategy.loadBalancer.allowedSourceRanges set to a valid CIDR range %s", validCIDR)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer.AllowedSourceRanges = []operatorv1.CIDR{operatorv1.CIDR(validCIDR)}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
```

Comment:
```suggestion
	t.Cleanup(assertIngressControllerDeleted(t, kclient, ic))
```

Reply:
Have we had a discussion about using `t.Cleanup`?  It looks like in this case, it makes no difference vis-à-vis using `defer`, and generally speaking, `t.Cleanup` is useful for test helpers, so maybe it makes sense to adopt it as a convention for all new tests.  

@frobware, you've mentioned `t.Cleanup` in the past; any concerns?